pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -353,13 +353,35 @@ func customNewProcess(pid int32) (*process.Process, error) {
 	return p, nil
 }
 
+// ignoreComms is a list of process names (matched against /proc/PID/comm) to
+// never report as a service.
+var ignoreComms = map[string]struct{}{
+	"sshd":             {},
+	"dhclient":         {},
+	"systemd":          {},
+	"systemd-resolved": {},
+	"systemd-networkd": {},
+	"datadog-agent":    {},
+	"livenessprobe":    {},
+	"docker-proxy":     {},
+}
+
 // getService gets information for a single service.
 func (s *discovery) getService(context parsingContext, pid int32) *model.Service {
 	proc, err := customNewProcess(pid)
 	if err != nil {
 		return nil
 	}
 
+	comm, err := proc.Name()
+	if err != nil {
+		return nil
+	}
+
+	if _, found := ignoreComms[comm]; found {
+		return nil
+	}
```

Comment:
The current code will work as all entries in `ignoreComms` are less than  or equal 16 characters
I do suspect we might add another entry to `ignoreComms` in the future, and we will not be aware of the assumption of 16 characters
I do suggest adding a comment & a UT to verify the entries length does not exceed 16 characters

Reply:
Done.